server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -467,10 +488,27 @@ public void uploadMetadata(
             );
             try {
                 IndexOutput indexOutput = storeDirectory.createOutput(metadataFilename, IOContext.DEFAULT);
+                List<SegmentCommitInfo> infos = segmentInfosSnapshot.asList();
```

Comment:
nit: Do we need this list only to iterate? If yes, we can use 
```for (SegmentCommitInfo segmentCommitInfo : segmentInfosSnapshot)```

Reply:
Would `segmentInfosSnapshot.asList()` return list with size greater than 1? We can add assertions to validate the size and create a corresponding IT if required.

Reply:
@ashking94 The existing `RemoteSegmentStoreDirectoryTests` validate the list of files

Reply:
Resolving this, please feel free to re-open if any update is needed.
Thanks!

Reply:
Can you point where we are validating the same? Thanks!